solutions/src/main/java/ru/sberbank/school/task02/Client.java

```diff
@@ -0,0 +1,108 @@
+package ru.sberbank.school.task02;
+
+import ru.sberbank.school.task02.exception.ConverterConfigurationException;
+import ru.sberbank.school.task02.exception.FxConversionException;
+import ru.sberbank.school.task02.exception.WrongSymbolException;
+import ru.sberbank.school.task02.util.*;
+
+import java.math.BigDecimal;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+
+public class Client implements FxClientController {
+    private ExtendedFxConversionService converter;
+    private String beneficiary;
+
+    public Client(ExternalQuotesService externalQuotesService) {
+        this.beneficiary = System.getenv("SBRF_BENEFICIARY");
+        ServiceFactoryImpl factory = new ServiceFactoryImpl();
+        this.converter = factory.getExtendedFxConversionService(externalQuotesService);
+    }
+
+    public static void main(String[] args) {
+        if (args.length < 3) {
+            throw new ConverterConfigurationException("Not enough input parameters");
+        }
+
+        ExternalQuotesService externalQuotesService = new ExternalQuotesServiceDemo();
+
+        Client client = new Client(externalQuotesService);
+
+        FxRequest request = new FxRequest(args[0], args[1], args[2]);
+
+        FxResponse response = client.fetchResult(request);
+
+        System.out.println(response);
+        System.out.println(response.getPrice());
+
+    }
+
+    @Override
+    public List<FxResponse> fetchResult(List<FxRequest> requests) {
+        List<FxResponse> responses = new ArrayList<>();
+
+        for (FxRequest request : requests) {
+            responses.add(fetchResult(request));
+        }
+        return responses;
+    }
+
+    @Override
+    public FxResponse fetchResult(FxRequest requests) {
+        ClientOperation operation = getOperation(requests.getDirection());
+        Symbol symbol = getSymbol(requests.getSymbol());
+        BigDecimal amount = new BigDecimal(requests.getAmount());
+
+        Optional<BigDecimal> price = beneficiary == null ?
```

Comment:
&#39;?&#39; should be on a new line.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a852978718>*